### PR TITLE
Opportunities in currency other than org default not auto closing

### PR DESCRIPTION
**Pull request by mdeleon0**
# Critical Changes
NO
# Changes
Modified class PMT_PaymentCreator.cls
Added unit tests in PMT_PaymentCreator_TEST.cls 
# Issues Closed
https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x80000007fORlAAM/view
# New Metadata
NO
# Deleted Metadata
NO

### CODE REVIEW

File: src/classes/PMT_PaymentCreator_TEST.cls [hunk 1]
```diff
@@ -1033,4 +1033,141 @@ private with sharing class PMT_PaymentCreator_TEST {
...
+        //create test data
+        Account acct = new Account (
+            Name='Test Account'
+        );
+        insert acct;
+
+        // I have an opportunity for CAD100 in a stage of Prospecting.
+        Opportunity opp = new Opportunity (
+            Name='Non US Currency',
+            AccountId = acct.id,
```
Comment (lmeerkatz) [inline src/classes/PMT_PaymentCreator_TEST.cls]: id -> Id

File: src/classes/PMT_PaymentCreator_TEST.cls [hunk 2]
```diff
@@ -1033,4 +1033,141 @@ private with sharing class PMT_PaymentCreator_TEST {
...
+
+        UTIL_CustomSettingsFacade.getContactsSettings().Payments_Auto_Close_Stage_Name__c = closedWonStage;
+
+        //create test data
+        Account acct = new Account (
+            Name='Test Account'
+        );
+        insert acct;
+
+        // I have an opportunity for CAD100 in a stage of Prospecting.
```
Comment (lmeerkatz) [inline src/classes/PMT_PaymentCreator_TEST.cls]: remove reference to CAD - we can't guarantee which currency we'll be in here

File: src/classes/PMT_PaymentCreator_TEST.cls [hunk 3]
```diff
@@ -1033,4 +1033,141 @@ private with sharing class PMT_PaymentCreator_TEST {
...
+        Opportunity opp = new Opportunity (
+            Name='Non US Currency',
+            AccountId = acct.id,
+            Amount = 100.00,
+            CloseDate = date.today(),
```
Comment (lmeerkatz) [inline src/classes/PMT_PaymentCreator_TEST.cls]: date -> Date

File: src/classes/PMT_PaymentCreator_TEST.cls [hunk 4]
```diff
@@ -1033,4 +1033,141 @@ private with sharing class PMT_PaymentCreator_TEST {
...
+            StageName = openStage
+        );
+
+        insert opp;
+
+        npe01__OppPayment__c payment = [SELECT id,
```
Comment (lmeerkatz) [inline src/classes/PMT_PaymentCreator_TEST.cls]: Fix capitalization on fields (Id, npe01__Paid__c, npe01__Payment_Amount__c)

File: src/classes/PMT_PaymentCreator_TEST.cls [hunk 5]
```diff
@@ -1033,4 +1033,141 @@ private with sharing class PMT_PaymentCreator_TEST {
...
+                                               npe01__paid__c,
+                                               npe01__payment_amount__c,
+                                               npe01__Payment_Date__c
+                                        FROM npe01__OppPayment__c
+                                        WHERE npe01__opportunity__c = :opp.Id
+                                        LIMIT 1];
+
+        payment.npe01__paid__c = true;
+        payment.npe01__Payment_Date__c = date.today();
```
Comment (lmeerkatz) [inline src/classes/PMT_PaymentCreator_TEST.cls]: date -> Date

File: src/classes/PMT_PaymentCreator_TEST.cls [hunk 6]
```diff
@@ -1033,4 +1033,141 @@ private with sharing class PMT_PaymentCreator_TEST {
...
+
+        UTIL_CustomSettingsFacade.getContactsSettings().Payments_Auto_Close_Stage_Name__c = closedWonStage;
+
+        //create test data
+        Account acct = new Account (
+            Name='Test Account'
+        );
+        insert acct;
+
+        // I have an opportunity for CAD100 in a stage of Prospecting.
```
Comment (lmeerkatz) [inline src/classes/PMT_PaymentCreator_TEST.cls]: Remove reference to CAD

File: src/classes/PMT_PaymentCreator_TEST.cls (hunk 1 above)
Comment (mdeleon0) [reply src/classes/PMT_PaymentCreator_TEST.cls]: Done.

File: src/classes/PMT_PaymentCreator_TEST.cls (hunk 2 above)
Comment (mdeleon0) [reply src/classes/PMT_PaymentCreator_TEST.cls]: Removed CAD from comment.

File: src/classes/PMT_PaymentCreator_TEST.cls (hunk 3 above)
Comment (mdeleon0) [reply src/classes/PMT_PaymentCreator_TEST.cls]: Changed.

File: src/classes/PMT_PaymentCreator_TEST.cls (hunk 5 above)
Comment (mdeleon0) [reply src/classes/PMT_PaymentCreator_TEST.cls]: Changed

File: src/classes/PMT_PaymentCreator_TEST.cls (hunk 6 above)
Comment (mdeleon0) [reply src/classes/PMT_PaymentCreator_TEST.cls]: Removed CAD from comment.

File: src/classes/PMT_PaymentCreator_TEST.cls (hunk 4 above)
Comment (mdeleon0) [reply src/classes/PMT_PaymentCreator_TEST.cls]: Fixed.

**Comment by CarmeNima** [comment]
QA has Passed.

**Comment by davisagli** [comment]
@sam-knox @smarovich this needs some work on the release notes (apologies if it's already on your radar)

@mdeleon0 in the future you can make things easiest for the docs team if you write the pull request description like this:
* if there's no content for one of the sections, leave it blank or remove it rather than writing "NO"
* if it's a fix for an issue, link to the github issue (not Agile Accelerator) like this: #xxx where xxx is the issue number. In this case you probably don't need to put anything in the Changes section unless there's something to highlight for our customers
* if it's a new or changed feature instead of a fix, in the Changes section describe the change from a customer perspective, rather than listing what components were edited. The docs team will edit this for the final release notes.

Thanks!

**Comment by sam-knox** [comment]
@mdeleon0 - I've updated the Changes section. Can you confirm that what I have there is technically correct? 

@smarovich

**Comment by lmeerkatz** [comment]
**lurch:attach W-028941

**Pull request merged**